internal/infrastructure/kubernetes/configmap.go

```diff
@@ -0,0 +1,179 @@
+package kubernetes
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"text/template"
+
+	corev1 "k8s.io/api/core/v1"
+	kerrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+
+	"github.com/envoyproxy/gateway/internal/ir"
+)
+
+var (
+	// envoyGatewayService is the name of the Envoy Gateway service.
+	envoyGatewayService = "envoy-gateway"
+)
+
+var envoyTmpl = template.Must(template.New(envoyCfgFileName).Parse(`
+admin:
+  access_log_path: /dev/null
+  address:
+    socket_address:
+      address: 127.0.0.1
+      port_value: 19000
+dynamic_resources:
+  cds_config:
+    resource_api_version: V3
+    api_config_source:
+      api_type: GRPC
+      transport_api_version: V3
+      grpc_services:
+      - envoy_grpc:
+          cluster_name: xds_cluster
+      set_node_on_first_message_only: true
+  lds_config:
+    resource_api_version: V3
+    api_config_source:
+      api_type: GRPC
+      transport_api_version: V3
+      grpc_services:
+      - envoy_grpc:
+          cluster_name: xds_cluster
+      set_node_on_first_message_only: true
+node:
+  cluster: envoy-gateway-system
+  id: envoy-default
+static_resources:
+  clusters:
+  - connect_timeout: 1s
+    load_assignment:
+      cluster_name: xds_cluster
+      endpoints:
+      - lb_endpoints:
+        - endpoint:
+            address:
+              socket_address:
+                address: {{ .XdsServerAddress }}
+                port_value: 18000
+    http2_protocol_options: {}
+    name: xds_cluster
+    type: STRICT_DNS
+layered_runtime:
+  layers:
+    - name: runtime-0
+      rtds_layer:
+        rtds_config:
+          resource_api_version: V3
+          api_config_source:
+            transport_api_version: V3
+            api_type: GRPC
+            grpc_services:
+              envoy_grpc:
+                cluster_name: xds_cluster
+        name: runtime-0
+`))
+
+// envoyConfigMap defines the managed Envoy ConfigMap object.
+type envoyConfigMap struct {
+	Key   string
+	Envoy envoyConfig
+}
+
+// envoyConfig contains Envoy configuration parameters.
+type envoyConfig struct {
+	// XdsServerAddress is the address of the XDS Server that Envoy is managed by.
+	XdsServerAddress string
+}
+
+// createConfigMapIfNeeded creates a ConfigMap based on the provided infra, if
```

Comment:
afaik a pod doesn't restart if the config map is updated, we'd still need extra instrumentation to ensure that the pod restarts when the config map changes
a simpler solution might be to use the [config-yaml](https://www.envoyproxy.io/docs/envoy/latest/operations/cli#cmdoption-config-yaml) argument and pass the config as a yaml string directly

Reply:
@arkodg I refactored the PR to use `config-yaml` instead of a ConfigMap. @LukeShu @youngnick https://github.com/envoyproxy/gateway/pull/211 is a draft PR to use the configmap approach. This approach requires add'l refactoring that I'd prefer doing after #197. In the meantime, I would like to proceed with the [config-yaml](https://www.envoyproxy.io/docs/envoy/latest/operations/cli#cmdoption-config-yaml) approach so we can begin e2e.